authors/apps/authentication/serializers.py

```diff
@@ -143,3 +145,51 @@ def update(self, instance, validated_data):
         instance.save()
 
         return instance
+
+
+class ForgotPasswordSerializer(serializers.ModelSerializer):
+    """Performs email serialization."""
+    email = serializers.EmailField()
+
+
+class ResetPasswordSerializers(serializers.ModelSerializer):
+    """
+    Performs reset password fields serialization.
+    """
+    email = serializers.EmailField(max_length=255)
+    password = serializers.CharField(
+        max_length=128,
+        min_length=8
+    )
+    confirm_password = serializers.CharField(
+        max_length=128,
+        min_length=8
+    )
+
+    token = serializers.CharField(max_length=255)
+
+    def validate(self, data):
+        """
+        Validates passwords match and
+        token can only be used once
+        """
+
+        # Query user from DB using email
+        email = data.get('email')
+        user = User.objects.filter(email).first()
+
+        if data.get('password') != data.get('confirm_password'):
+            msg = "Passwords do not match"
+            raise serializers.ValidationError(msg)
+
+        # Confirm if token is valid
+        token = data.get('token')
+        validate_token = default_token_generator.check_token(user, token)
```

Comment:
the token will be generated in the view

Reply:
are you using token generated by Django

Reply:
Yes using `PasswordResetTokenGenerator` this class handles password reset tokens

Reply:
I think just my opinion @charisschomba 💭  ...if` default_token_generator` can check for a `token`, it should `generate that token`...right? just confirm if it exists because how does it confirm a token that it hasn't generated, you know like in flask , werkzeung it `generate password hash` and then check `check_password_hash`

Reply:
it does generate a token @YAHYA-H

Reply:
okay, cool if it works 😄 💯...